src/Persistence/MassTransit.Azure.Storage/Configuration/AzureStorageConfigurationExtensions.cs

```diff
@@ -1,14 +1,13 @@
 namespace MassTransit.Azure.Storage
 {
     using MessageData;
-    using Microsoft.Azure.Storage;
-
+    using global::Azure.Storage.Blobs;
 
     public static class AzureStorageConfigurationExtensions
     {
-        public static AzureStorageMessageDataRepository CreateMessageDataRepository(this CloudStorageAccount account, string containerName)
+        public static AzureStorageMessageDataRepository CreateMessageDataRepository(this BlobServiceClient account, string containerName)
```

Comment:
I'm not in favor of changing the signature, since it would break anyone using the current client. Is it fully backwards compatible? Meaning that if I have a storage account full of message data will the new version be able to read it? Have you tested this?

Reply:
> [..] if I have a storage account full of message data will the new version be able to read it?
> Have you tested this?

Yes it is backwards compatible. I have tested this scenario and it passed.
I have an integration test that essentially runs the repository implementations side-by-side; in this integration test I `Put` the message-data using the existing implementation and `Get` the message-data using the new implementation.
Hmm I should do the reverse too, I'll do that tomorrow.

> I'm not in favor of changing the signature, since it would break anyone using the current client. 

I agree and was reluctant to do so; unfortunately to my understanding it is not possible to avoid the signature changing if we drop the deprecated package.
This is because the new package shipped from Microsoft - `Azure.Storage.Blobs` - no longer contains the class `CloudStorageAccount` (and several others).

I'll do further research tomorrow to see if I've missed something that would allow for no source changes without referencing a deprecated package.

Reply:
Nice work on the integration testing, well done!

Reply:
Thanks, the test fixtures in project's testing framework are very nice to use for all this!

I did the reverse scenario; that passed too.